graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java

```diff
@@ -0,0 +1,149 @@
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OpenSearchExceptionTest {
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    Response response;
+
+    @Test
+    public void handle429() throws IOException {
+        when(response.getHost()).thenReturn(new HttpHost("myHost"));
+
+        RequestLine requestLine = new RequestLine() {
+            @Override
+            public String getMethod() {
+                return "POST";
+            }
+
+            @Override
+            public ProtocolVersion getProtocolVersion() {
+                return null;
+            }
+
+            @Override
+            public String getUri() {
+                return "http://myTest.com";
+            }
+        };
+        when(response.getRequestLine()).thenReturn(requestLine);
+
+        StatusLine statusLine = new StatusLine() {
+            @Override
+            public ProtocolVersion getProtocolVersion() {
+                return null;
+            }
+
+            @Override
+            public int getStatusCode() {
+                return 429;
+            }
+
+            @Override
+            public String getReasonPhrase() {
+                return null;
+            }
+        };
+        when(response.getStatusLine()).thenReturn(statusLine);
+
+        HttpEntity httpEntity = new HttpEntity() {
+            @Override
+            public boolean isRepeatable() {
+                return false;
+            }
+
+            @Override
+            public boolean isChunked() {
+                return false;
+            }
+
+            @Override
+            public long getContentLength() {
+                return 0;
+            }
+
+            @Override
+            public Header getContentType() {
+                return null;
+            }
+
+            @Override
+            public Header getContentEncoding() {
+                return null;
+            }
+
+            @Override
+            public InputStream getContent() throws IOException, UnsupportedOperationException {
+                return null;
+            }
+
+            @Override
+            public void writeTo(OutputStream outputStream) throws IOException {
+
+            }
+
+            @Override
+            public boolean isStreaming() {
+                return false;
+            }
+
+            @Override
+            public void consumeContent() throws IOException {
+
+            }
+        };
+        when(response.getEntity()).thenReturn(httpEntity);
+
+        ResponseException responseException = new ResponseException(response);
+        RestStatus restStatus = RestStatus.BAD_REQUEST;
+        OpenSearchStatusException statusException = new OpenSearchStatusException(
+                "status msg", restStatus, responseException);
+
+        OpenSearchInstance openSearchInstance = OpenSearchInstance.create();
+        final OpenSearchClient openSearchClient = openSearchInstance.openSearchClient();
```

Comment:
```suggestion
        final OpenSearchClient openSearchClient = new OpenSearchClient(mock(RestHighLevelClient.class), true, new ObjectMapper());```
        
If we are going to throw the exception ourselves anyway, we don't need a real opensearch instance, which takes minutes to start, but can also just mock the client.

Reply:
also we should've called
`openSearchInstance.close()` to shut down the test instance properly

Reply:
Good point. Removed the instance.